src/consensus/aft/raft.h

```diff
@@ -466,6 +469,14 @@ namespace aft
         become_retired(idx, kv::RetirementPhase::Ordered);
       }
 
+#ifdef CCF_RAFT_TRACING
+      nlohmann::json j = {};
+      j["function"] = "add_configuration";
+      j["state"] = *state;
+      j["configurations"] = configurations;
```

Comment:
The behavior has changed, causing `configurations` to be empty `[]` initially (see `replicate.ndjson`).

Reply:
By the way, I'm not claiming that the new behavior is wrong.  However, it violates the third conjunct of [`ccfraft.tla! InitReconfigurationVars `](https://github.com/microsoft/CCF/blob/b5e27b0213df527b3814ab758be8e225d9e4f19b/tla/ccfraft.tla#L537-L543).

Reply:
The implementation hasn't changed behaviour as such, but https://github.com/microsoft/CCF/compare/main...lemmy:CCF:mku-gh5057#diff-4e61c67c4990874cc31cb6f8cd91e4abe7506492fce0f73a4705d556d6861a42R493 was moving forward (with a different condition) the update to configurations when the new config differs the last active.

Ideally we would just move the trace further down, but I think there are a number of actions we take based on this that should be logically logged later. I'll try to reorganise the code without breaking anything.

Reply:
@lemmy it would be easier to log this earlier if we split `configurations` and `config`/`new_config`. The spec seems to do that anyway on lines 252-253. I'll make that change if that's ok.

Reply:
Clarification question:  Will this result in *two* consecutive log lines?

Reply:
No, it's still a single line, I've updated the traces. But this can be changed! :)

Reply:
No reason, one line is fine!